### PR TITLE
Segmentio tracking for inventory API.

### PR DESCRIPTION
**Pull request by roliveri**
Initial POC for adding instrumentation to track inventory API events in Segmentio.

### CODE REVIEW

File: api/segmentio.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+            identity.org_id,
+            op_name,
+            {
+                "start_time": start_time,
+                "end_time": end_time,
+                "status_code": contextual_data["status_code"],
+                "user_agent": connexion.request.headers["User-Agent"],
+                "auth_type": identity.auth_type,
+            },
+        )
```
[the remark this answers is not in the document]
Comment (roliveri) [reply api/segmentio.py:23]: Yes, I'll add that.
We should probably check to see if there's a standard of preferred name for insights host inventory.

File: api/segmentio.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+def segmentio_track(op_name, start_time, end_time, contextual_data):
+    identity = get_current_identity()
+
+    print(f"\n**** analytics.write_key: {analytics.write_key}")
+    if analytics.write_key:
+        print(f"**** Calling analytics.track({identity.org_id}, {op_name}, ...)")
+        analytics.track(
+            identity.org_id,
+            op_name,
```
Comment (marleystipich2) [inline api/segmentio.py:15]: Small suggestion but maybe we could name these 2 fields too for readabilty like
```
user_id=identity.org_id,
event=op_name,
```

File: api/segmentio.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
+import connexion
+import segment.analytics as analytics
+
+from app.auth import get_current_identity
+
+
+def segmentio_track(op_name, start_time, end_time, contextual_data):
+    identity = get_current_identity()
+
+    print(f"\n**** analytics.write_key: {analytics.write_key}")
```
Comment (thearifismail) [inline api/segmentio.py:10]: Should't a `logger` be used in place of `print`?

File: app/__init__.py [hunk 4]
```diff
@@ -85,6 +86,14 @@ def initialize_metrics(config):
     )
 
 
+# TODO: Need to call analytics.flush() on shutdown.
+def initialize_segmentio(config):
+    # Should this be maintained by app/config.py?
+    analytics.write_key = config.segmentio_write_key
+    print(f"\n****** Initializing Segmentio - analytics.write_key: {analytics.write_key}")
```
Comment (thearifismail) [inline app/__init__.py:93]: Should use logger in place of print

**Review by thearifismail — COMMENTED** [review]
Why another analytic tracking tool is needed on the top of what's already implemented?

**Comment by thearifismail** [comment]
> @kruai @thearifismail We also need to call `analytics.flush()` on shutdown. I looked, but I'm not sure where that should be done. Any suggestions?

How will `analytics` start or initialize?  The invenvetory MQ service uses ShutdownHandler, which is in `lib/handlers.py`

File: api/segmentio.py (hunk 3 above)
Comment (kruai) [reply api/segmentio.py:10]: You're probably referring to the other print statements, but just in case: We should absolutely not print or log secrets

**Review by thearifismail — COMMENTED** [review]
One comment; rest LGTM

**Comment by marleystipich2** [comment]
@thearifismail We added some tests to this. Could we get another review when you get a chance? Also, there are some tests failing in the PR check but they pass locally.

**Review by kruai — CHANGES_REQUESTED** [review]

**Review by kruai — APPROVED** [review]

**Review by marleystipich2 — APPROVED** [review]
Looks good to me! Thanks, @roliveri 👍

**Review by thearifismail — APPROVED** [review]

**Pull request merged**